lib/riskified/request.rb

```
@@ -14,19 +14,25 @@ def initialize(resource, json_body)
     end
 
     def send
-      begin
-        request = Typhoeus::Request.new(
-            endpoint,
-            method: :post,
-            body: @json_body,
-            headers: prepare_headers
-        )
+      request = Typhoeus::Request.new(
+          endpoint,
+          method: :post,
+          body: @json_body,
+          headers: prepare_headers
+      )
 
-        response = request.run
-      rescue StandardError => e
-        raise Riskified::Exceptions::ApiConnectionError.new(e.message)
+      request.on_complete do |response|
+        if response.timed_out?
+          raise Riskified::Exceptions::ApiConnectionError.new("Unable to connect. Request was timed out.")
```

i think you might want to throw the error in a way that somebody can catch it more precisely. When the code that is using this gem wants to catch timeouts, the only way to do so is to read the text 'Unable to connect. Request was timed out.'. Reading that text is a bad practice. Therefore:
 - make a specific error class (eg ApiTimeoutError) that extends from ApiConnectionError or
 - add error codes. In your ApiConnectionError you could declare a few constants (eg TIMEOUT_ERROR = 1) and add that to the exception Riskified::Exceptions::ApiConnectionError.new("Unable to connect. Request was timed out.", ApiConnectionError::TIMEOUT_ERROR)

---

I've adde a custom error, but didn't add the custom error codes as there is no much exceptions in the gem anyway.